Boost broken on OS X CI build

https://github.com/offa/influxdb-cxx/actions/runs/17364587151/job/49289436561?pr=316

```
 -- Conan: Including build module from '/Users/runner/.conan2/p/b/opensccde6d969f7af/p/lib/cmake/conan-official-openssl-variables.cmake'
CMake Error at /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_system"
-- Configuring incomplete, errors occurred!
  (requested version 1.89.0) with any of the following names:

    boost_systemConfig.cmake
    boost_system-config.cmake

  Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
  "boost_system_DIR" to a directory containing one of the above files.  If
  "boost_system" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
  CMakeLists.txt:93 (find_package)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,7 +90,7 @@ if (INFLUXCXX_WITH_BOOST)
     # Fixes warning when using boost from brew
     set(Boost_USE_MULTITHREADED TRUE)
 
-    find_package(Boost REQUIRED COMPONENTS system)
+    find_package(Boost REQUIRED)
 endif()
 
 add_subdirectory(3rd-party)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -14,7 +14,7 @@ target_include_directories(InfluxDB-BoostSupport PRIVATE ${INTERNAL_INCLUDE_DIRS
 target_link_libraries(InfluxDB-BoostSupport PRIVATE date::date)
 
 if (INFLUXCXX_WITH_BOOST)
-    target_link_libraries(InfluxDB-BoostSupport PRIVATE Boost::boost Boost::system)
+    target_link_libraries(InfluxDB-BoostSupport PRIVATE Boost::boost)
 endif()
 
 # #117: Workaround for Boost ASIO null-dereference

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -46,7 +46,7 @@ target_sources(NoBoostSupportTest PRIVATE ${PROJECT_SOURCE_DIR}/src/NoBoostSuppo
 target_link_libraries(NoBoostSupportTest PRIVATE InfluxDB)
 
 if (INFLUXCXX_WITH_BOOST)
-    add_unittest(BoostSupportTest DEPENDS InfluxDB-BoostSupport InfluxDB Boost::system date::date)
+    add_unittest(BoostSupportTest DEPENDS InfluxDB-BoostSupport InfluxDB Boost::boost date::date)
 endif()
 
 

